In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/content/global_sports_footwear_sales_2018_2026.csv')

In [3]:
df

,order_id,order_date,brand,model_name,category,gender,size,color,base_price_usd,discount_percent,final_price_usd,units_sold,revenue_usd,payment_method,sales_channel,country,customer_income_level,customer_rating
0,ORD100000,2021-01-30,ASICS,Model-370,Running,Unisex,8,Black,162,15,137.70,1,137.70,Card,Retail Store,Germany,Low,4.6
1,ORD100001,2026-10-05,Reebok,Model-314,Lifestyle,Men,8,Grey,80,5,76.00,3,228.00,Card,Online,USA,Low,3.9
2,ORD100002,2023-11-12,ASICS,Model-763,Lifestyle,Men,8,Black,176,15,149.60,4,598.40,Cash,Retail Store,India,Medium,3.0
3,ORD100003,2026-08-29,Reebok,Model-905,Basketball,Women,7,White,61,15,51.85,2,103.70,Card,Retail Store,India,High,3.4
4,ORD100004,2019-11-09,Nike,Model-413,Training,Men,11,Black,80,0,80.00,4,320.00,Cash,Online,USA,Medium,3.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
29995,ORD129995,2018-05-28,Puma,Model-892,Gym,Women,6,Blue,140,10,126.00,1,126.00,Card,Online,UAE,High,4.3
29996,ORD129996,2018-04-28,Adidas,Model-982,Running,Women,9,Grey,113,15,96.05,1,96.05,Card,Online,UK,High,3.5
29997,ORD129997,2026-02-17,ASICS,Model-463,Basketball,Men,11,Blue,130,30,91.00,1,91.00,Wallet,Online,India,Medium,4.4
29998,ORD129998,2024-02-21,New Balance,Model-984,Running,Unisex,11,White,186,10,167.40,2,334.80,Wallet,Retail Store,USA,High,3.9


In [4]:
df.shape

(30000, 18)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   order_id               30000 non-null  object 
 1   order_date             30000 non-null  object 
 2   brand                  30000 non-null  object 
 3   model_name             30000 non-null  object 
 4   category               30000 non-null  object 
 5   gender                 30000 non-null  object 
 6   size                   30000 non-null  int64  
 7   color                  30000 non-null  object 
 8   base_price_usd         30000 non-null  int64  
 9   discount_percent       30000 non-null  int64  
 10  final_price_usd        30000 non-null  float64
 11  units_sold             30000 non-null  int64  
 12  revenue_usd            30000 non-null  float64
 13  payment_method         30000 non-null  object 
 14  sales_channel          30000 non-null  object 
 15  co

In [6]:
df.describe()

,size,base_price_usd,discount_percent,final_price_usd,units_sold,revenue_usd,customer_rating
count,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000
mean,8.500867,139.634633,13.332167,121.029035,2.500200,302.714948,4.001543
std,1.710896,46.062549,9.864198,42.511586,1.121149,179.149272,0.577546
min,6.000000,60.000000,0.000000,42.000000,1.000000,42.000000,3.000000
25%,7.000000,100.000000,5.000000,85.000000,1.000000,156.750000,3.500000
50%,9.000000,140.000000,10.000000,119.200000,3.000000,268.200000,4.000000
75%,10.000000,180.000000,20.000000,153.600000,4.000000,414.000000,4.500000
max,11.000000,219.000000,30.000000,219.000000,4.000000,876.000000,5.000000


In [7]:
df['order_date'] = pd.to_datetime(df['order_date'])

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   order_id               30000 non-null  object        
 1   order_date             30000 non-null  datetime64[ns]
 2   brand                  30000 non-null  object        
 3   model_name             30000 non-null  object        
 4   category               30000 non-null  object        
 5   gender                 30000 non-null  object        
 6   size                   30000 non-null  int64         
 7   color                  30000 non-null  object        
 8   base_price_usd         30000 non-null  int64         
 9   discount_percent       30000 non-null  int64         
 10  final_price_usd        30000 non-null  float64       
 11  units_sold             30000 non-null  int64         
 12  revenue_usd            30000 non-null  float64       
 13  p

In [9]:
df = df.sort_values(by = 'order_date', ascending = False).reset_index()

In [10]:
df = df.drop('index', axis = 1)

In [11]:
df

,order_id,order_date,brand,model_name,category,gender,size,color,base_price_usd,discount_percent,final_price_usd,units_sold,revenue_usd,payment_method,sales_channel,country,customer_income_level,customer_rating
0,ORD112082,2026-12-31,ASICS,Model-588,Lifestyle,Unisex,8,Blue,154,0,154.00,3,462.00,Wallet,Retail Store,UAE,Medium,3.8
1,ORD112997,2026-12-31,Reebok,Model-376,Lifestyle,Unisex,8,White,167,0,167.00,4,668.00,Wallet,Online,Germany,High,4.3
2,ORD106780,2026-12-31,New Balance,Model-426,Training,Men,11,Grey,207,30,144.90,4,579.60,Bank Transfer,Retail Store,UAE,Medium,3.6
3,ORD121066,2026-12-31,Puma,Model-926,Training,Men,7,White,126,10,113.40,4,453.60,Card,Online,Germany,Medium,4.4
4,ORD101905,2026-12-31,Nike,Model-290,Running,Men,10,Red,123,5,116.85,1,116.85,Bank Transfer,Online,UK,High,3.9
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
29995,ORD107881,2018-01-01,ASICS,Model-674,Lifestyle,Women,9,White,217,15,184.45,1,184.45,Bank Transfer,Online,USA,Medium,3.0
29996,ORD105845,2018-01-01,Puma,Model-874,Running,Unisex,7,Grey,165,10,148.50,1,148.50,Wallet,Retail Store,USA,High,4.6
29997,ORD121309,2018-01-01,Adidas,Model-512,Running,Women,10,White,191,5,181.45,1,181.45,Wallet,Online,USA,Low,5.0
29998,ORD114642,2018-01-01,New Balance,Model-927,Gym,Women,10,Red,117,0,117.00,1,117.00,Cash,Online,Pakistan,High,4.4


In [12]:
df.groupby('order_date').size().agg(['min', 'max'])

,0
min,1
max,21


In [13]:
df.groupby('order_date').size()

,0
order_date,
2018-01-01,6
2018-01-02,12
2018-01-03,11
2018-01-04,13
2018-01-05,7
...,...
2026-12-27,9
2026-12-28,9
2026-12-29,12


In [14]:
df.groupby('brand').sum('final_price_usd')

,size,base_price_usd,discount_percent,final_price_usd,units_sold,revenue_usd,customer_rating
brand,,,,,,,
ASICS,43596,718120,68270,622417.75,12874,1561462.50,20573.9
Adidas,41719,687327,66290,594371.35,12326,1486266.55,19613.3
New Balance,42380,693692,66110,601453.20,12541,1511401.50,19920.2
Nike,42466,701838,66190,609538.20,12481,1524582.10,20063.8
Puma,41968,690619,66670,597582.15,12315,1499094.90,19708.8
Reebok,42897,697443,66435,605508.40,12469,1498640.90,20166.3


In [15]:
df.groupby(by = ['brand', 'category'])[['revenue_usd','units_sold']].sum().sort_values(by= 'revenue_usd', ascending = False)

revenue_usd  units_sold
brand       category                           
ASICS       Lifestyle     330533.70        2696
New Balance Training      322023.05        2673
Puma        Lifestyle     318943.60        2584
Nike        Training      314562.25        2590
            Basketball    312283.10        2572
ASICS       Training      310020.45        2523
Reebok      Running       309787.15        2548
ASICS       Basketball    309239.85        2611
New Balance Running       308533.05        2546
ASICS       Gym           308507.10        2533
Puma        Basketball    304181.10        2502
Adidas      Training      303434.65        2499
            Basketball    303302.80        2449
ASICS       Running       303161.40        2511
Adidas      Lifestyle     300987.10        2485
Nike        Lifestyle     300541.80        2464
            Gym           300273.65        2424
New Balance Basketball    298548.70        2505
Reebok      Training      298151.05        2484
            Lifestyle     298063.00        2467
            Gym           297825.60        2449
Nike        Running       296921.30        2431
New Balance Lifestyle     295559.55        2433
Puma        Running       295246.95        2422
Reebok      Basketball    294814.10        2521
Puma        Gym           292576.05        2434
Adidas      Running       291800.85        2470
Puma        Training      288147.20        2373
Adidas      Gym           286741.15        2423
New Balance Gym           286737.15        2384

In [16]:
df.groupby(by = ['brand', 'category'])[['revenue_usd','units_sold']].sum().sort_values(by= 'units_sold', ascending = False)

,,revenue_usd,units_sold
brand,category,,
ASICS,Lifestyle,330533.70,2696
New Balance,Training,322023.05,2673
ASICS,Basketball,309239.85,2611
Nike,Training,314562.25,2590
Puma,Lifestyle,318943.60,2584
Nike,Basketball,312283.10,2572
Reebok,Running,309787.15,2548
New Balance,Running,308533.05,2546
ASICS,Gym,308507.10,2533


In [17]:
top_by_units = df.groupby(by=['brand', 'category'])[['revenue_usd', 'units_sold']].sum().sort_values(by='units_sold', ascending=False)
top_by_revenue = df.groupby(by=['brand', 'category'])[['revenue_usd', 'units_sold']].sum().sort_values(by='revenue_usd', ascending=False)

print(top_by_units.head(10))
print(top_by_revenue.head(10))

                        revenue_usd  units_sold
brand       category                           
ASICS       Lifestyle     330533.70        2696
New Balance Training      322023.05        2673
ASICS       Basketball    309239.85        2611
Nike        Training      314562.25        2590
Puma        Lifestyle     318943.60        2584
Nike        Basketball    312283.10        2572
Reebok      Running       309787.15        2548
New Balance Running       308533.05        2546
ASICS       Gym           308507.10        2533
            Training      310020.45        2523
                        revenue_usd  units_sold
brand       category                           
ASICS       Lifestyle     330533.70        2696
New Balance Training      322023.05        2673
Puma        Lifestyle     318943.60        2584
Nike        Training      314562.25        2590
            Basketball    312283.10        2572
ASICS       Training      310020.45        2523
Reebok      Running       309787.15     

**Insight: Brand and category performance is unusually balanced across the dataset**

Grouping sales by brand and category shows no single combination dominating — revenue across the top 10 groups ranges narrowly between ~305,000 and 330,000 USD, and units sold cluster between ~2,500 and 2,700. ASICS (Lifestyle) and New Balance (Training) lead marginally in both revenue and units, but the rankings shift slightly further down the list depending on whether you sort by units sold or revenue — for example, Puma (Lifestyle) and Nike (Training) outrank ASICS (Basketball) on revenue despite having fewer or similar total units, suggesting modestly higher per-unit pricing in those categories.

That said, the overall uniformity across nearly every brand/category pairing is itself the more notable finding — real-world sales data typically shows more pronounced winners and laggards. This suggests the dataset was generated with intentionally balanced distributions across brands and categories, rather than reflecting organic, skewed market demand. As a result, brand/category alone isn't a strong lever for differentiating performance in this dataset — worth checking whether other dimensions (country, sales channel, income level) show clearer patterns.

In [18]:
df['discount_bucket'] = pd.cut(df['discount_percent'], bins = [-1,0,10,20,100], labels = ['No Discount','1-10%', '11-20%', '21%+'])

In [19]:
df.groupby(by = 'discount_bucket')[['units_sold','revenue_usd']].mean()

/tmp/ipykernel_14695/2025491262.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby(by = 'discount_bucket')[['units_sold','revenue_usd']].mean()


,units_sold,revenue_usd
discount_bucket,,
No Discount,2.487628,346.773889
1-10%,2.501739,323.988438
11-20%,2.498645,288.187293
21%+,2.512692,245.078913


**Insight: Discounting does not meaningfully increase purchase volume**

Grouping orders by discount level shows average units sold per order stays nearly constant regardless of discount size (2.49 units at no discount vs. 2.51 units at 21%+ discount — a negligible difference). Meanwhile, average revenue per order declines steadily as discount increases, from ~ $347 at no discount to ~ $245 at 21%+ discount, a drop of roughly 29%. This indicates that within this dataset, discounting functions purely as a revenue reducer rather than a volume driver — customers do not appear to purchase more units in response to larger discounts. Combined with the earlier finding that this dataset shows unusually uniform performance across brand and category, this reinforces that the data likely reflects synthetically generated patterns rather than organic consumer price sensitivity, where discounting would typically show at least some correlation with purchase volume.

In [20]:
df.head()

,order_id,order_date,brand,model_name,category,gender,size,color,base_price_usd,discount_percent,final_price_usd,units_sold,revenue_usd,payment_method,sales_channel,country,customer_income_level,customer_rating,discount_bucket
0,ORD112082,2026-12-31,ASICS,Model-588,Lifestyle,Unisex,8,Blue,154,0,154.00,3,462.00,Wallet,Retail Store,UAE,Medium,3.8,No Discount
1,ORD112997,2026-12-31,Reebok,Model-376,Lifestyle,Unisex,8,White,167,0,167.00,4,668.00,Wallet,Online,Germany,High,4.3,No Discount
2,ORD106780,2026-12-31,New Balance,Model-426,Training,Men,11,Grey,207,30,144.90,4,579.60,Bank Transfer,Retail Store,UAE,Medium,3.6,21%+
3,ORD121066,2026-12-31,Puma,Model-926,Training,Men,7,White,126,10,113.40,4,453.60,Card,Online,Germany,Medium,4.4,1-10%
4,ORD101905,2026-12-31,Nike,Model-290,Running,Men,10,Red,123,5,116.85,1,116.85,Bank Transfer,Online,UK,High,3.9,1-10%


In [21]:
pd.crosstab(df['country'], df['sales_channel'])

sales_channel,Online,Retail Store
country,,
Germany,2503,2462
India,2484,2507
Pakistan,2387,2454
UAE,2562,2556
UK,2537,2521
USA,2518,2509


In [23]:
df.groupby(by = ['sales_channel','country'])[['revenue_usd','units_sold']].sum()

revenue_usd  units_sold
sales_channel country                          
Online        Germany     767643.40        6309
              India       744964.95        6164
              Pakistan    726768.15        5986
              UAE         779252.70        6456
              UK          770066.25        6326
              USA         751933.75        6286
Retail Store  Germany     736251.45        6143
              India       775934.00        6375
              Pakistan    739396.45        6077
              UAE         767189.85        6362
              UK          762233.80        6336
              USA         759813.70        6186

In [24]:
df.groupby(['country', 'sales_channel'])['units_sold'].sum().unstack()

sales_channel,Online,Retail Store
country,,
Germany,6309,6143
India,6164,6375
Pakistan,5986,6077
UAE,6456,6362
UK,6326,6336
USA,6286,6186


In [25]:
channel_totals = df.groupby(['country', 'sales_channel'])['units_sold'].sum().unstack()
channel_totals['online_pct'] = channel_totals['Online'] / (channel_totals['Online'] + channel_totals['Retail Store']) * 100
print(channel_totals)

sales_channel  Online  Retail Store  online_pct
country                                        
Germany          6309          6143   50.666560
India            6164          6375   49.158625
Pakistan         5986          6077   49.622814
UAE              6456          6362   50.366672
UK               6326          6336   49.960512
USA              6286          6186   50.400898



**Insight: Sales channel split shows no meaningful variation by country**

Analyzing the online vs. retail split by country shows every market sitting within roughly 1 percentage point of an even 50/50 split (ranging from 49.2% online in India to 50.7% in Germany). In real-world footwear retail, online adoption typically varies more substantially by country due to differences in e-commerce infrastructure, logistics, and consumer trust — markets like India or Pakistan often show a stronger retail-store skew compared to the US, UK, or Germany. The near-uniform split observed here, consistent with the earlier findings on brand/category performance and discount response, further supports that this dataset was generated with evenly balanced synthetic distributions rather than reflecting organic regional consumer behavior.

In [26]:
df

,order_id,order_date,brand,model_name,category,gender,size,color,base_price_usd,discount_percent,final_price_usd,units_sold,revenue_usd,payment_method,sales_channel,country,customer_income_level,customer_rating,discount_bucket
0,ORD112082,2026-12-31,ASICS,Model-588,Lifestyle,Unisex,8,Blue,154,0,154.00,3,462.00,Wallet,Retail Store,UAE,Medium,3.8,No Discount
1,ORD112997,2026-12-31,Reebok,Model-376,Lifestyle,Unisex,8,White,167,0,167.00,4,668.00,Wallet,Online,Germany,High,4.3,No Discount
2,ORD106780,2026-12-31,New Balance,Model-426,Training,Men,11,Grey,207,30,144.90,4,579.60,Bank Transfer,Retail Store,UAE,Medium,3.6,21%+
3,ORD121066,2026-12-31,Puma,Model-926,Training,Men,7,White,126,10,113.40,4,453.60,Card,Online,Germany,Medium,4.4,1-10%
4,ORD101905,2026-12-31,Nike,Model-290,Running,Men,10,Red,123,5,116.85,1,116.85,Bank Transfer,Online,UK,High,3.9,1-10%
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
29995,ORD107881,2018-01-01,ASICS,Model-674,Lifestyle,Women,9,White,217,15,184.45,1,184.45,Bank Transfer,Online,USA,Medium,3.0,11-20%
29996,ORD105845,2018-01-01,Puma,Model-874,Running,Unisex,7,Grey,165,10,148.50,1,148.50,Wallet,Retail Store,USA,High,4.6,1-10%
29997,ORD121309,2018-01-01,Adidas,Model-512,Running,Women,10,White,191,5,181.45,1,181.45,Wallet,Online,USA,Low,5.0,1-10%
29998,ORD114642,2018-01-01,New Balance,Model-927,Gym,Women,10,Red,117,0,117.00,1,117.00,Cash,Online,Pakistan,High,4.4,No Discount


In [27]:
df.groupby(by = 'customer_income_level')[['final_price_usd']].mean()

,final_price_usd
customer_income_level,
High,120.553554
Low,121.855727
Medium,120.685156


In [28]:
pd.crosstab(df['customer_income_level'],df['category'])

category,Basketball,Gym,Lifestyle,Running,Training
customer_income_level,,,,,
High,2069,1953,2040,1954,2043
Low,1982,1973,1990,2045,1954
Medium,2023,1940,2029,1984,2021


In [29]:
pd.crosstab(df['customer_income_level'],df['brand'])

brand,ASICS,Adidas,New Balance,Nike,Puma,Reebok
customer_income_level,,,,,,
High,1685,1634,1708,1727,1627,1678
Low,1744,1662,1647,1594,1649,1648
Medium,1703,1622,1627,1697,1648,1700


In [33]:
df.groupby(by = ['customer_income_level','discount_bucket'])['units_sold'].mean()

/tmp/ipykernel_14695/3324323235.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby(by = ['customer_income_level','discount_bucket'])['units_sold'].mean()


customer_income_level  discount_bucket
High                   No Discount        2.491187
                       1-10%              2.494731
                       11-20%             2.508069
                       21%+               2.531953
Low                    No Discount        2.495663
                       1-10%              2.528235
                       11-20%             2.482159
                       21%+               2.547962
Medium                 No Discount        2.476133
                       1-10%              2.481747
                       11-20%             2.505789
                       21%+               2.461001
Name: units_sold, dtype: float64

**Insight: Customer income level does not predict purchasing behavior**

Testing income level against average price paid, category preference, brand preference, and discount sensitivity all show the same result: no meaningful variation. Average final price ranges only $1.31 across income levels ($120.55–$121.86), category and brand distributions are evenly spread within roughly 5% of each other across all income groups, and discount responsiveness (average units sold per order) shows no consistent pattern by income level, with all values clustering tightly between 2.46 and 2.55 regardless of discount size. In real consumer markets, income level typically predicts at least some purchasing behavior — such as price sensitivity or brand preference. Its complete absence here, combined with the earlier findings on brand/category, discount response, and channel/country splits, provides strong, consistent evidence that this dataset was synthetically generated with uniformly randomized distributions across all customer segments, rather than modeling real consumer behavior.

In [34]:
df['order_year'] = df['order_date'].dt.year
df['order_month'] = df['order_date'].dt.month

In [35]:
df.groupby('order_year')[['revenue_usd', 'units_sold']].sum()

,revenue_usd,units_sold
order_year,,
2018,990380.20,8181
2019,1029728.15,8576
2020,1032345.75,8494
2021,986065.65,8142
2022,1026659.90,8469
2023,1025135.90,8385
2024,970039.25,8043
2025,1024096.85,8483
2026,996996.80,8233


In [36]:
df.groupby('order_month')[['revenue_usd', 'units_sold']].sum()

,revenue_usd,units_sold
order_month,,
1,771040.40,6409
2,699173.70,5782
3,751808.65,6205
4,752074.50,6192
5,778805.40,6376
6,748830.55,6290
7,783344.95,6488
8,756421.05,6236
9,750702.90,6244


**Insight: No meaningful trend across time — confirming the dataset is synthetic**

Yearly revenue shows no growth or decline pattern across 2018–2026, fluctuating narrowly between $970K and $1.03M with no discernible trajectory. Notably, 2020 posts the highest revenue in the dataset ($1.03M) rather than a dip, despite this being a year when real-world footwear retail was significantly disrupted by COVID-19 store closures — a strong indicator the dates were randomly assigned rather than modeled on real market conditions. Monthly totals are similarly flat, with no visible holiday-season spike in November/December that real footwear sales typically show. Combined with the earlier findings across brand, category, discount response, geography, sales channel, and income level — all of which showed no meaningful variation — this provides comprehensive evidence that the dataset was generated synthetically with uniformly randomized values across every dimension tested, rather than reflecting real consumer purchasing behavior.

In [38]:
df.to_csv('global_sports_footwear_sales_cleaned.csv', index = False)